Add support for TPM530M module and TPM530M EVK board

tokangas opened the pull request:
Added SoC support for Sercomm TPM530M module and  board configuration for TPM530M EVK.

NCSDK-27893

tokangas committed:
soc: sercomm: Add support for TPM530M

Added support for Sercomm TPM530M.

Signed-off-by: Tommi Kangas <tommi.kangas@nordicsemi.no>

tejlmand reviewed (commented):
Looks good, couple of minor observations.

NordicBuilder commented:
## CI Information

<sup>_To view the history of this post, clich the 'edited' button above_</sup>
Build number: 1
### Inputs:
#### Sources:
sdk-nrf: PR head: [4ab142b0450e408dc70733d54125f2be8341d85b](https://github.com/nrfconnect/sdk-nrf/pull/23858/commits/4ab142b0450e408dc70733d54125f2be8341d85b)
<details><summary> more details</summary>

sdk-nrf:
> PR head: [4ab142b0450e408dc70733d54125f2be8341d85b](https://github.com/nrfconnect/sdk-nrf/pull/23858/commits/4ab142b0450e408dc70733d54125f2be8341d85b)
> merge base: [fbe765ac2d592166c0a4c79539cd14f81036cb64](https://github.com/nrfconnect/sdk-nrf/commit/fbe765ac2d592166c0a4c79539cd14f81036cb64)
> target head (main): [7c54925b7f4304b45ea9c9305780762aa86da077](https://github.com/nrfconnect/sdk-nrf/commit/7c54925b7f4304b45ea9c9305780762aa86da077)
> [Diff](https://github.com/nrfconnect/sdk-nrf/compare/fbe765ac2d592166c0a4c79539cd14f81036cb64..4ab142b0450e408dc70733d54125f2be8341d85b)</details>

#### Github labels
| Enabled | Name | Description |
| -- | -- | -- |
|| ci-disabled | _Disable the ci execution_ | 
|| ci-all-test | _Run all of ci, no test spec filtering will be done_ | 
|| ci-force-downstream | _Force execution of downstream even if twister fails_ | 
|| ci-run-twister | _Force run twister_ | 
|| ci-run-zephyr-twister | _Force run zephyr twister_ | 

<details><summary> List of changed files detected by CI (22)</summary>

```
boards
│  ├── sercomm
│  │  ├── index.rst
│  │  ├── tpm530mevk
│  │  │  ├── Kconfig.defconfig
│  │  │  ├── Kconfig.tpm530mevk
│  │  │  ├── board.cmake
│  │  │  ├── board.yml
│  │  │  ├── doc
│  │  │  │  │ index.rst
│  │  │  ├── pre_dt_board.cmake
│  │  │  ├── tpm530mevk_tpm530m.dts
│  │  │  ├── tpm530mevk_tpm530m.yaml
│  │  │  ├── tpm530mevk_tpm530m_common-pinctrl.dtsi
│  │  │  ├── tpm530mevk_tpm530m_common.dtsi
│  │  │  ├── tpm530mevk_tpm530m_defconfig
│  │  │  ├── tpm530mevk_tpm530m_ns.dts
│  │  │  ├── tpm530mevk_tpm530m_ns.yaml
│  │  │  │ tpm530mevk_tpm530m_ns_defconfig
dts
│  ├── arm
│  │  ├── sercomm
│  │  │  ├── tpm530m.dtsi
│  │  │  │ tpm530mns.dtsi
soc
│  ├── sercomm
│  │  ├── nrf91
│  │  │  ├── CMakeLists.txt
│  │  │  ├── Kconfig.defconfig
│  │  │  ├── Kconfig.defconfig.tpm530m
│  │  │  ├── Kconfig.soc
│  │  │  │ soc.yml
```
</details>

### Outputs:

#### Toolchain
Version: 
Build docker image: 

#### Test Spec & Results: :white_check_mark: Success; :x: Failure; :orange_circle: Queued; :yellow_circle: Progress; :white_medium_square: Skipped; :warning: Quarantine
- :orange_circle: Toolchain
- :orange_circle: Build twister
- :orange_circle: Integration tests

<details><summary>Disabled integration tests</summary>

- 
   - test-fw-nrfconnect-nrf_lrcs_mosh
   - test-fw-nrfconnect-nrf_lrcs_positioning
   - desktop52_verification
   - doc-internal
   - test_ble_nrf_config
   - test-fw-nrfconnect-apps
   - test-fw-nrfconnect-ble_mesh
   - test-fw-nrfconnect-ble_samples
   - test-fw-nrfconnect-chip
   - test-fw-nrfconnect-fem
   - test-fw-nrfconnect-nfc
   - test-fw-nrfconnect-nrf-iot_cloud
   - test-fw-nrfconnect-nrf-iot_libmodem-nrf
   - test-fw-nrfconnect-nrf-iot_lwm2m
   - test-fw-nrfconnect-nrf-iot_samples
   - test-fw-nrfconnect-nrf-iot_serial_lte_modem
   - test-fw-nrfconnect-nrf-iot_thingy91
   - test-fw-nrfconnect-nrf-iot_zephyr_lwm2m
   - test-fw-nrfconnect-nrf_crypto
   - test-fw-nrfconnect-ps-main
   - test-fw-nrfconnect-rpc
   - test-fw-nrfconnect-rs
   - test-fw-nrfconnect-tfm
   - test-fw-nrfconnect-thread-main
   - test-low-level
   - test-sdk-audio
   - test-sdk-dfu
   - test-sdk-find-my
   - test-sdk-mcuboot
   - test-sdk-pmic-samples
   - test-sdk-wifi
   - test-secdom-samples-public
</details>



*Note: This message is automatically posted and updated by the CI*

NordicBuilder commented:
### Memory footprint analysis revealed the following potential issues

***applications.nrf_desktop.zdebug.uart[nrf54l15dk/nrf54l10/cpuapp]***: ***ROM*** size  ***increased by 1696[B]*** in comparison to the ***main[2f3e5f4b]*** branch. - [link](https://ci-health.nordicsemi.no/grafana/d/footprint-tracking-sample/footprint-tracking-for-samples-and-applications-sdk-nrf?orgId=1&var-module=applications&var-submodule=nrf_desktop&var-scenario=zdebug&var-testcase=uart&var-arch=arm&var-platform=nrf54l15dk/nrf54l10/cpuapp) (cc: @nrfconnect/ncs-si-bluebagel)
***applications.nrf_desktop.zdebug.uart[nrf54l15dk/nrf54l15/cpuapp]***: ***ROM*** size  ***increased by 1696[B]*** in comparison to the ***main[2f3e5f4b]*** branch. - [link](https://ci-health.nordicsemi.no/grafana/d/footprint-tracking-sample/footprint-tracking-for-samples-and-applications-sdk-nrf?orgId=1&var-module=applications&var-submodule=nrf_desktop&var-scenario=zdebug&var-testcase=uart&var-arch=arm&var-platform=nrf54l15dk/nrf54l15/cpuapp) (cc: @nrfconnect/ncs-si-bluebagel)
***applications.nrf_desktop.zdebug.uart[nrf54lm20dk/nrf54lm20a/cpuapp]***: ***ROM*** size  ***increased by 1704[B]*** in comparison to the ***main[2f3e5f4b]*** branch. - [link](https://ci-health.nordicsemi.no/grafana/d/footprint-tracking-sample/footprint-tracking-for-samples-and-applications-sdk-nrf?orgId=1&var-module=applications&var-submodule=nrf_desktop&var-scenario=zdebug&var-testcase=uart&var-arch=arm&var-platform=nrf54lm20dk/nrf54lm20a/cpuapp) (cc: @nrfconnect/ncs-si-bluebagel)
***applications.nrf_desktop.zdebug.uart[nrf54lm20pdk@0.2.0.csp/nrf54lm20a/cpuapp]***: ***ROM*** size  ***increased by 1696[B]*** in comparison to the ***main[2f3e5f4b]*** branch. - [link](https://ci-health.nordicsemi.no/grafana/d/footprint-tracking-sample/footprint-tracking-for-samples-and-applications-sdk-nrf?orgId=1&var-module=applications&var-submodule=nrf_desktop&var-scenario=zdebug&var-testcase=uart&var-arch=arm&var-platform=nrf54lm20pdk@0.2.0.csp/nrf54lm20a/cpuapp) (cc: @nrfconnect/ncs-si-bluebagel)
***applications.nrf_desktop.zdebug.uart[nrf54lm20pdk@0.2.0/nrf54lm20a/cpuapp]***: ***ROM*** size  ***increased by 1696[B]*** in comparison to the ***main[2f3e5f4b]*** branch. - [link](https://ci-health.nordicsemi.no/grafana/d/footprint-tracking-sample/footprint-tracking-for-samples-and-applications-sdk-nrf?orgId=1&var-module=applications&var-submodule=nrf_desktop&var-scenario=zdebug&var-testcase=uart&var-arch=arm&var-platform=nrf54lm20pdk@0.2.0/nrf54lm20a/cpuapp) (cc: @nrfconnect/ncs-si-bluebagel)
***applications.nrf_desktop.zdebug.usb_next[nrf52840dk/nrf52840]***: ***ROM*** size  ***increased by 1536[B]*** in comparison to the ***main[2f3e5f4b]*** branch. - [link](https://ci-health.nordicsemi.no/grafana/d/footprint-tracking-sample/footprint-tracking-for-samples-and-applications-sdk-nrf?orgId=1&var-module=applications&var-submodule=nrf_desktop&var-scenario=zdebug&var-testcase=usb_next&var-arch=arm&var-platform=nrf52840dk/nrf52840) (cc: @nrfconnect/ncs-si-bluebagel)
***applications.nrf_desktop.zdebug.usb_next[nrf52840gmouse/nrf52840]***: ***ROM*** size  ***increased by 1660[B]*** in comparison to the ***main[2f3e5f4b]*** branch. - [link](https://ci-health.nordicsemi.no/grafana/d/footprint-tracking-sample/footprint-tracking-for-samples-and-applications-sdk-nrf?orgId=1&var-module=applications&var-submodule=nrf_desktop&var-scenario=zdebug&var-testcase=usb_next&var-arch=arm&var-platform=nrf52840gmouse/nrf52840) (cc: @nrfconnect/ncs-si-bluebagel)
***applications.nrf_desktop.zdebug[nrf52840dk/nrf52840]***: ***ROM*** size  ***increased by 1648[B]*** in comparison to the ***main[2f3e5f4b]*** branch. - [link](https://ci-health.nordicsemi.no/grafana/d/footprint-tracking-sample/footprint-tracking-for-samples-and-applications-sdk-nrf?orgId=1&var-module=applications&var-submodule=nrf_desktop&var-scenario=zdebug&var-testcase=none&var-arch=arm&var-platform=nrf52840dk/nrf52840) (cc: @nrfconnect/ncs-si-bluebagel)
***applications.nrf_desktop.zdebug[nrf52840gmouse/nrf52840]***: ***ROM*** size  ***increased by 1676[B]*** in comparison to the ***main[2f3e5f4b]*** branch. - [link](https://ci-health.nordicsemi.no/grafana/d/footprint-tracking-sample/footprint-tracking-for-samples-and-applications-sdk-nrf?orgId=1&var-module=applications&var-submodule=nrf_desktop&var-scenario=zdebug&var-testcase=none&var-arch=arm&var-platform=nrf52840gmouse/nrf52840) (cc: @nrfconnect/ncs-si-bluebagel)
***applications.nrf_desktop.zdebug[nrf52dmouse/nrf52832]***: ***ROM*** size  ***increased by 1336[B]*** in comparison to the ***main[2f3e5f4b]*** branch. - [link](https://ci-health.nordicsemi.no/grafana/d/footprint-tracking-sample/footprint-tracking-for-samples-and-applications-sdk-nrf?orgId=1&var-module=applications&var-submodule=nrf_desktop&var-scenario=zdebug&var-testcase=none&var-arch=arm&var-platform=nrf52dmouse/nrf52832) (cc: @nrfconnect/ncs-si-bluebagel)
***applications.nrf_desktop.zdebug[nrf52kbd/nrf52832]***: ***ROM*** size  ***increased by 1320[B]*** in comparison to the ***main[2f3e5f4b]*** branch. - [link](https://ci-health.nordicsemi.no/grafana/d/footprint-tracking-sample/footprint-tracking-for-samples-and-applications-sdk-nrf?orgId=1&var-module=applications&var-submodule=nrf_desktop&var-scenario=zdebug&var-testcase=none&var-arch=arm&var-platform=nrf52kbd/nrf52832) (cc: @nrfconnect/ncs-si-bluebagel)
***applications.nrf_desktop.zdebug[nrf54l15dk/nrf54l05/cpuapp]***: ***ROM*** size  ***increased by 940[B]*** in comparison to the ***main[2f3e5f4b]*** branch. - [link](https://ci-health.nordicsemi.no/grafana/d/footprint-tracking-sample/footprint-tracking-for-samples-and-applications-sdk-nrf?orgId=1&var-module=applications&var-submodule=nrf_desktop&var-scenario=zdebug&var-testcase=none&var-arch=arm&var-platform=nrf54l15dk/nrf54l05/cpuapp) (cc: @nrfconnect/ncs-si-bluebagel)
***applications.nrf_desktop.zdebug_fast_pair.gmouse.uart[nrf54l15dk/nrf54l10/cpuapp]***: ***ROM*** size  ***increased by 1644[B]*** in comparison to the ***main[2f3e5f4b]*** branch. - [link](https://ci-health.nordicsemi.no/grafana/d/footprint-tracking-sample/footprint-tracking-for-samples-and-applications-sdk-nrf?orgId=1&var-module=applications&var-submodule=nrf_desktop&var-scenario=zdebug_fast_pair&var-testcase=gmouse.uart&var-arch=arm&var-platform=nrf54l15dk/nrf54l10/cpuapp) (cc: @nrfconnect/ncs-si-bluebagel)
***applications.nrf_desktop.zdebug_fast_pair.gmouse.uart[nrf54l15dk/nrf54l15/cpuapp]***: ***ROM*** size  ***increased by 1644[B]*** in comparison to the ***main[2f3e5f4b]*** branch. - [link](https://ci-health.nordicsemi.no/grafana/d/footprint-tracking-sample/footprint-tracking-for-samples-and-applications-sdk-nrf?orgId=1&var-module=applications&var-submodule=nrf_desktop&var-scenario=zdebug_fast_pair&var-testcase=gmouse.uart&var-arch=arm&var-platform=nrf54l15dk/nrf54l15/cpuapp) (cc: @nrfconnect/ncs-si-bluebagel)
***applications.nrf_desktop.zdebug_fast_pair.gmouse[nrf52840gmouse/nrf52840]***: ***ROM*** size  ***increased by 1568[B]*** in comparison to the ***main[2f3e5f4b]*** branch. - [link](https://ci-health.nordicsemi.no/grafana/d/footprint-tracking-sample/footprint-tracking-for-samples-and-applications-sdk-nrf?orgId=1&var-module=applications&var-submodule=nrf_desktop&var-scenario=zdebug_fast_pair&var-testcase=gmouse&var-arch=arm&var-platform=nrf52840gmouse/nrf52840) (cc: @nrfconnect/ncs-si-bluebagel)
***applications.nrf_desktop.zdebug_keyboard.uart[nrf54l15dk/nrf54l10/cpuapp]***: ***ROM*** size  ***increased by 1532[B]*** in comparison to the ***main[2f3e5f4b]*** branch. - [link](https://ci-health.nordicsemi.no/grafana/d/footprint-tracking-sample/footprint-tracking-for-samples-and-applications-sdk-nrf?orgId=1&var-module=applications&var-submodule=nrf_desktop&var-scenario=zdebug_keyboard&var-testcase=uart&var-arch=arm&var-platform=nrf54l15dk/nrf54l10/cpuapp) (cc: @nrfconnect/ncs-si-bluebagel)
***applications.nrf_desktop.zdebug_keyboard.uart[nrf54l15dk/nrf54l15/cpuapp]***: ***ROM*** size  ***increased by 1532[B]*** in comparison to the ***main[2f3e5f4b]*** branch. - [link](https://ci-health.nordicsemi.no/grafana/d/footprint-tracking-sample/footprint-tracking-for-samples-and-applications-sdk-nrf?orgId=1&var-module=applications&var-submodule=nrf_desktop&var-scenario=zdebug_keyboard&var-testcase=uart&var-arch=arm&var-platform=nrf54l15dk/nrf54l15/cpuapp) (cc: @nrfconnect/ncs-si-bluebagel)
***applications.nrf_desktop.zdebug_keyboard.usb_next[nrf52840dk/nrf52840]***: ***ROM*** size  ***increased by 920[B]*** in comparison to the ***main[2f3e5f4b]*** branch. - [link](https://ci-health.nordicsemi.no/grafana/d/footprint-tracking-sample/footprint-tracking-for-samples-and-applications-sdk-nrf?orgId=1&var-module=applications&var-submodule=nrf_desktop&var-scenario=zdebug_keyboard&var-testcase=usb_next&var-arch=arm&var-platform=nrf52840dk/nrf52840) (cc: @nrfconnect/ncs-si-bluebagel)
***applications.nrf_desktop.zdebug_keyboard[nrf52840dk/nrf52840]***: ***ROM*** size  ***increased by 932[B]*** in comparison to the ***main[2f3e5f4b]*** branch. - [link](https://ci-health.nordicsemi.no/grafana/d/footprint-tracking-sample/footprint-tracking-for-samples-and-applications-sdk-nrf?orgId=1&var-module=applications&var-submodule=nrf_desktop&var-scenario=zdebug_keyboard&var-testcase=none&var-arch=arm&var-platform=nrf52840dk/nrf52840) (cc: @nrfconnect/ncs-si-bluebagel)
***applications.nrf_desktop.zdebug_mcuboot_qspi[nrf52840dk/nrf52840]***: ***ROM*** size  ***increased by 1652[B]*** in comparison to the ***main[2f3e5f4b]*** branch. - [link](https://ci-health.nordicsemi.no/grafana/d/footprint-tracking-sample/footprint-tracking-for-samples-and-applications-sdk-nrf?orgId=1&var-module=applications&var-submodule=nrf_desktop&var-scenario=zdebug_mcuboot_qspi&var-testcase=none&var-arch=arm&var-platform=nrf52840dk/nrf52840) (cc: @nrfconnect/ncs-si-bluebagel)
***applications.nrf_desktop.zdebug_mcuboot_smp[nrf52840dk/nrf52840]***: ***ROM*** size  ***increased by 1632[B]*** in comparison to the ***main[2f3e5f4b]*** branch. - [link](https://ci-health.nordicsemi.no/grafana/d/footprint-tracking-sample/footprint-tracking-for-samples-and-applications-sdk-nrf?orgId=1&var-module=applications&var-submodule=nrf_desktop&var-scenario=zdebug_mcuboot_smp&var-testcase=none&var-arch=arm&var-platform=nrf52840dk/nrf52840) (cc: @nrfconnect/ncs-si-bluebagel)
***applications.nrf_desktop.zdebug_nrf21540ek[nrf52840dk/nrf52840]***: ***ROM*** size  ***increased by 1652[B]*** in comparison to the ***main[2f3e5f4b]*** branch. - [link](https://ci-health.nordicsemi.no/grafana/d/footprint-tracking-sample/footprint-tracking-for-samples-and-applications-sdk-nrf?orgId=1&var-module=applications&var-submodule=nrf_desktop&var-scenario=zdebug_nrf21540ek&var-testcase=none&var-arch=arm&var-platform=nrf52840dk/nrf52840) (cc: @nrfconnect/ncs-si-bluebagel)
***applications.nrf_desktop.zdebug_wwcb[nrf52840dk/nrf52840]***: ***ROM*** size  ***increased by 1556[B]*** in comparison to the ***main[2f3e5f4b]*** branch. - [link](https://ci-health.nordicsemi.no/grafana/d/footprint-tracking-sample/footprint-tracking-for-samples-and-applications-sdk-nrf?orgId=1&var-module=applications&var-submodule=nrf_desktop&var-scenario=zdebug_wwcb&var-testcase=none&var-arch=arm&var-platform=nrf52840dk/nrf52840) (cc: @nrfconnect/ncs-si-bluebagel)
***applications.nrf_desktop.zdebugwithshell[nrf52840dk/nrf52840]***: ***ROM*** size  ***increased by 1640[B]*** in comparison to the ***main[2f3e5f4b]*** branch. - [link](https://ci-health.nordicsemi.no/grafana/d/footprint-tracking-sample/footprint-tracking-for-samples-and-applications-sdk-nrf?orgId=1&var-module=applications&var-submodule=nrf_desktop&var-scenario=zdebugwithshell&var-testcase=none&var-arch=arm&var-platform=nrf52840dk/nrf52840) (cc: @nrfconnect/ncs-si-bluebagel)
***applications.nrf_desktop.zrelease[nrf52kbd/nrf52832]***: ***ROM*** size  ***increased by 620[B]*** in comparison to the ***main[2f3e5f4b]*** branch. - [link](https://ci-health.nordicsemi.no/grafana/d/footprint-tracking-sample/footprint-tracking-for-samples-and-applications-sdk-nrf?orgId=1&var-module=applications&var-submodule=nrf_desktop&var-scenario=zrelease&var-testcase=none&var-arch=arm&var-platform=nrf52kbd/nrf52832) (cc: @nrfconnect/ncs-si-bluebagel)
***applications.nrf_desktop.zrelease_fast_pair.keyboard[nrf52kbd/nrf52832]***: ***ROM*** size  ***increased by 620[B]*** in comparison to the ***main[2f3e5f4b]*** branch. - [link](https://ci-health.nordicsemi.no/grafana/d/footprint-tracking-sample/footprint-tracking-for-samples-and-applications-sdk-nrf?orgId=1&var-module=applications&var-submodule=nrf_desktop&var-scenario=zrelease_fast_pair&var-testcase=keyboard&var-arch=arm&var-platform=nrf52kbd/nrf52832) (cc: @nrfconnect/ncs-si-bluebagel)


*Note: This message is automatically posted and updated by the CI (latest/sdk-nrf/PR-23858/2)*

tokangas committed:
boards: sercomm: Add board configuration for TPM530M EVK

Added board configuration for TPM530M EVK.

Signed-off-by: Tommi Kangas <tommi.kangas@nordicsemi.no>

NordicBuilder commented:
The following west manifest projects have changed revision in this Pull Request:

| Name | Old Revision | New Revision | Diff |
| ---- | ------------ | ------------ |------|
| zephyr | https://github.com/nrfconnect/sdk-zephyr/commit/8d7b9c9d00552a89eb04755435e24d555894d819 | https://github.com/nrfconnect/sdk-zephyr/pull/3186 | [nrfconnect/sdk-zephyr#3186/files](https://github.com/nrfconnect/sdk-zephyr/pull/3186/files) |

⛔ **DNM label due to: 1 project with PR revision**

*Note: This message is automatically posted and updated by the Manifest GitHub Action.*

tejlmand committed:
config: remove CONFIG_PINCTRL=y from defconfig files

CONFIG_PINCTRL=y fron defconfig files.
This is not the right location and should be removed.

Signed-off-by: Torsten Rasmussen <Torsten.Rasmussen@nordicsemi.no>

tokangas commented:
@nrfconnect/ncs-co-drivers @nrfconnect/ncs-merge @nrfconnect/ncs-co-boards @nrfconnect/ncs-cia @nrfconnect/ncs-si-bluebagel Please review.
